### ISSUE TITLE
Training Script

### ISSUE THREAD

**Issue by ELMIAR-0642** [amhummer/Graphinity#6]
I want to ask you, 

- When will you release the training script?
- Do you know if I can use the Data provided in the repo as my training data? (because we don't possess a large storage space to store all of the synthetic datasets)
- Can you release a mini data subset so that people can have a `lite` or `small` version of the model trained?
- Will you update the `README.md` with detailed explanations and instructions?
- The parameters of `train_dataloader` contain some `configs` to be passed in the form of the dictionary and `run_id`, what are those?
- Here are some questions related to Data:
    * How to use the splits/cutoffs provided in CSV? how to interpret them? 
    * Can I just use the `synthetic_ddg_wt_pdbs` dataset (2.6 GiB) alone, and train a baseline model, because I have space constraints? 
    * And I can't find the PDB IDs of the PDBs present in the `synthetic_ddg_wt_pdbs` dataset in the cutoffs CSV, why is that?
- There is not much explanation regarding the graph neural network methods(functions) from the scripts. 
- If I am writing a custom training script, for the time being, how should I proceed, and what should I include and not include?
- How to use the `train_dtaloader` method given in the ``egnn_model.py``?
- When and where to use `binder_classification` and `ddg_regression`?

**Comment by amhummer** [comment]
Hi @ELMIAR-0642, thank you for your interest in Graphinity!

Regarding your questions:

1. We will release the training code when the paper is published in a journal. I unfortunately do not have more information regarding a timeline for publication at this point.

2. The data in the repo links to the PDB files at the download links, so unfortunately cannot be used by itself. The repo does not include any PDBs (which are required to generate the graphs for the model input).

3. Certainly, I can look into creating a smaller subset of the dataset for easier downloading and will keep you posted.

4. I am planning to add more information to the `README` when the paper is published and I upload the training code. If there is any particular information you have questions about or would like me to include, please let me know.

5. The config file passes parameters to the model. These parameters are defined, many with default values given, in the respective .py files (classes and functions). I will release a sample config file together with the training code when the paper is published in a journal. The `run_id` parameter is deprecated/no longer used – I had used it in model development when saving files.

6. We applied a range of cutoffs based on CDR sequence identity to cluster the datasets. We then grouped the clusters into 10 folds of approximately equal size for 10-fold cross-validation. The `fold_id` column indicates which fold each data point belongs to. This can be split into train, validation, and test, for example: train = folds 0-7, validation = fold 8, test = fold 9. No antibody in one fold will have more than the cutoff CDR sequence identity with an antibody in another fold. I just noticed though that the `fold_id` column was missing from the synthetic `cdr_seqid_cutoffs` datasets – my apologies, I have updated this now.

7. The ddG architecture is intended for predicting the effects of mutations on affinity and thus requires as input PDBs of both the WT and mutated complexes.

8. Could you please give me some examples of PDBs present in the `synthetic_ddg_wt_pdbs` dataset but not in the cutoff CSVs, or vice versa? For my files, they match.

9. If you would like to read more about the graph neural network architecture, I would refer you to the publication: https://arxiv.org/abs/2102.09844. If you have any specific questions, please let me know.

10. The code is compatible with PyTorch Lightning. Regarding parameters, it depends which ones you want to keep at default values or change.

11. The `train_dataloader` in `egnn_model.py` calls the `ddgDataSet` / `dgDataSet` class from `dataset.py`. The populate function in the `(d)dgDataSet` class loads data based on a given csv file (containing the following columns: `labels`, `pdb_wt`, `pdb_mut`, `chain_prot1`, `chain_prot2`). The `(d)dgDataSet` loaded with a csv file is used as input for a `torch_geometric.data.DataLoader`.

12. The `binder_classification` method takes as input only one structure per data point with a corresponding binary label (eg binding or non-binding). The `ddg_regression` method takes as input two structures per data point (WT and mutant) with a corresponding quantitative label (ddG value). The choice of model depends on whether you are performing classification or regression, as well as your whether your structural input is one structure (affinity) or two (change in affinity).

**Comment by amhummer** [comment]
@ELMIAR-0642 I have now uploaded a smaller toy dataset (`Graphinity/data/ddg_synthetic/toy_dataset/`), which should be much easier to download. Please let me know if you have any questions.
